packages/wonder-blocks-floating/src/components/floating.tsx

```diff
@@ -1,5 +1,208 @@
 import * as React from "react";
+import {
+    useFloating,
+    autoUpdate,
+    offset,
+    flip,
+    hide,
+    shift,
+    arrow,
+    Placement,
+} from "@floating-ui/react";
+import {StyleSheet} from "aphrodite";
+import {
+    border,
+    boxShadow,
+    semanticColor,
+} from "@khanacademy/wonder-blocks-tokens";
+import {addStyle} from "@khanacademy/wonder-blocks-core";
+import {ARROW_SIZE_INLINE} from "../util/constants";
+import {Arrow} from "./floating-arrow";
 
-export default function Floating() {
-    return <div>Floating</div>;
+const StyledDiv = addStyle("div");
+
+type FloatingProps = {
+    /**
+     * The reference (or anchored) element that is used to calculate the
+     * position of the floating element.
+     */
+    children: React.ReactElement;
+    /**
+     * The content to display in the floating element.
+     */
+    content: React.ReactNode;
+
+    /**
+     * The padding to apply between the floating element and its boundary.
+     * @default 8
+     */
+    boundaryPadding?: number;
+
+    /**
+     * The placement of the floating element relative to the reference element.
+     * @default "top"
+     * @see https://floating-ui.com/docs/useFloating#placement
+     */
+    placement?: Placement;
+
+    /**
+     * The strategy to use for positioning the floating element.
+     * @default "absolute"
+     * @see https://floating-ui.com/docs/useFloating#strategy
+     */
+    strategy?: "fixed" | "absolute";
+
+    /**
+     * Whether the floating element is open.
+     * @default false
+     */
+    open: boolean;
+    /**
+     * Callback for when the floating element is opened or closed.
+     */
+    onOpenChange?: (open: boolean) => void;
+
+    /**
+     * The test ID to use for the floating element.
+     */
+    testId?: string;
+
+    // ----- Middleware specific props -----
+    /**
+     * The flip strategy to use.
+     * @default true
+     */
+    flip?: boolean;
+
+    /**
+     * Whether to hide the floating element when the reference element is hidden.
+     * @default true
+     */
+    hide?: boolean;
+
+    /**
+     * The offset of the floating element from the reference element.
+     * @default 20
+     */
+    offset?: number;
+
+    /**
+     * The shift strategy to use.
+     * @default true
+     */
+    shift?: boolean;
+
+    /**
+     * Whether to show the arrow on the floating element.
+     * @default true
+     */
+    showArrow?: boolean;
+
+    // TODO(WB-2111.3): Add props for dismissEnabled, portal and useFocusManager
+};
+
+/**
+ * A component that uses the Floating UI library to position a floating element
+ * relative to a reference element. The floating element appears on hover or
+ * focus.
```

Comment:
While going through the component docs, I read the line: "The floating element appears on hover or focus" right before trying out the first example, where the floating element appears on click! Should we include the clicking interaction in this sentence?

And should we include examples for showing the element on hover and on focus?